ipaserver/install/ipa_server_certinstall.py

```diff
@@ -161,6 +174,55 @@ def install_http_cert(self):
         os.chown(os.path.join(dirname, 'key3.db'), 0, pent.pw_gid)
         os.chown(os.path.join(dirname, 'secmod.db'), 0, pent.pw_gid)
 
+    def install_kdc_cert(self):
+        ca_cert_file = paths.CA_BUNDLE_PEM
+        pkcs12_file, pin, ca_cert = installutils.load_pkcs12(
+            cert_files=self.args,
+            key_password=self.options.pin,
+            key_nickname=self.options.cert_name,
+            ca_cert_files=[ca_cert_file],
+            realm_name=api.env.realm)
+
+        cdb = certs.CertDB(api.env.realm, nssdir=paths.IPA_NSSDB_DIR)
+
+        # Check that the ca_cert is known and trusted
+        with tempfile.NamedTemporaryFile() as temp:
+            certs.install_pem_from_p12(pkcs12_file.name, pin, temp.name)
+
+            kdc_cert = x509.load_certificate_from_file(temp.name)
+            ca_certs = x509.load_certificate_list_from_file(ca_cert_file)
+
+            try:
+                verify_kdc_cert_validity(kdc_cert, ca_certs, api.env.realm)
+            except ValueError as e:
+                raise admintool.ScriptError(
+                    "Peer's certificate issuer is not trusted (%s). "
+                    "Please run ipa-cacert-manage install and ipa-certupdate "
+                    "to install the CA certificate." % str(e))
```

Comment:
This is consistent with the code in `ServerCertInstall.check_chain()` and the lack of refactoring is intentional.